client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/HashBasedShuffleWriter.java

```diff
@@ -146,7 +146,7 @@ public HashBasedShuffleWriter(
           new DataPusher(
               shuffleId,
               mapId,
-              taskContext.attemptNumber(),
+              SparkUtils.getMapAttemptNumber(taskContext),
```

Comment:
Better to make `encodedAttemptId` as a member instead of calling `SparkUtils.getMapAttemptNumber(taskContext)` everywhere